core/dbt/include/global_project/macros/materializations/snapshot/strategies.sql

```diff
@@ -66,6 +66,7 @@
 {% macro snapshot_timestamp_strategy(node, snapshotted_rel, current_rel, config, target_exists) %}
     {% set primary_key = config['unique_key'] %}
     {% set updated_at = config['updated_at'] %}
+    {% set invalidate_hard_deletes = config['invalidate_hard_deletes'] %}
```

Comment:
I think this should be `config.get('invalidate_hard_deletes', false)` - `config` is a dict here, and we don't want to promise that `invalidate_hard_deletes` is present.

I think this works currently because dbt uses a non-strict `undefined`, which is false-ish. But I don't want to risk these `undefined` values escaping the context, they cause annoying json serialization issues down the line.

The other two configs are required, so they don't need to use `.get` and can use the regular `[...]` syntax.